app/uk/gov/hmrc/helptosavefrontend/controllers/UpdateEmailAddressController.scala

```diff
@@ -44,9 +49,29 @@ class UpdateEmailAddressController @Inject()(val sessionCacheConnector: SessionC
           } {
             _.eligibilityCheckResult.fold(
               Ok(views.html.core.not_eligible())
-            )(_ ⇒ Ok(views.html.register.update_email_address()))
+            )( userInfo ⇒ {
+              Ok(views.html.register.update_email_address(userInfo.contactDetails.email, Some(UpdateEmailForm.verifyEmailForm)))
+            })
           }
         }
   }(redirectOnLoginURL = FrontendAppConfig.checkEligibilityUrl)
 
-}
+  def onSubmit(): Action[AnyContent] = authorisedForHtsWithInfo {
+    implicit request =>
+      implicit htsContext ⇒
+        UpdateEmailForm.verifyEmailForm.bindFromRequest().fold(
```

Comment:
should probably do the `checkIfAlreadyEnrollled` and `checkSession` business here. This'd protect this endpoint. Also, it'll give you access to a non-optional NINO to use below